File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/CatalystRelationships.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable

/**
 * Relationship information between users
 */
@Serializable
data class CatalystRelationships(
    val isMyself: Boolean,
    val isFollowing: Boolean,
    val isFollowed: Boolean,
    val isBlocking: Boolean
)

/**
 * Following/follower counts for a user
 */
@Serializable
data class CatalystRelationshipsCount(
    val followings: Int? = null,
    val followers: Int? = null
)

/**
 * Paginated list of a user's followers or followings
 */
@Serializable
data class CatalystFollowingOrFollowersList(
    val items: List<EgeriaUser>,
    val count: CatalystCountInfo,
    val page: CatalystPageInfo
)

File: packages/swift/Sources/CatalystSwift/Models/CatalystRelationships.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

public struct CatalystRelationships: Decodable, Sendable, Equatable, Hashable {
  public let isMyself: Bool
  public let isFollowing: Bool
  public let isFollowed: Bool
  public let isBlocking: Bool
}

/// Following/follower counts for a user.
public struct CatalystRelationshipsCount: Decodable, Sendable, Equatable, Hashable {
  public let followings: Int?
  public let followers: Int?
}

/// Paginated list of a user's followers or followings.
public struct CatalystFollowingOrFollowersList: Decodable, Sendable {
  public let items: [EgeriaUser]
  public let count: CatalystCountInfo
  public let page: CatalystPageInfo
}
